library/src/main/java/org/apache/apex/malhar/lib/wal/FileSystemWAL.java

```diff
@@ -680,9 +681,7 @@ protected void flush() throws IOException
           //until the stream is closed on the local FS, readers don't see any data.
           close();
         } else {
-          Syncable syncableOutputStream = (Syncable)outputStream;
-          syncableOutputStream.hflush();
-          syncableOutputStream.hsync();
+          ((HdfsDataOutputStream)outputStream).hsync(EnumSet.of(SyncFlag.UPDATE_LENGTH));
```

Comment:
@sandeshh if you are referring to the use of length in readOrSkip, then this looks incorrect. The WAL needs to be self-sufficient and not rely on data that is stored elsewhere. This should be done by sequentially reading the file till there is no more data. Further, namenode operations should be avoided, especially when they occur on every streaming window. The flush/sync as it stands is sufficient to save the data and keep the file open, rest should be addressed when reading the WAL, which is the infrequent operation.